modules/matrix_controller.py

```
@@ -19,329 +17,477 @@
+                        preview_size = 16  # Pixels per LED
+                        preview = Image.new('RGB', (controller.W * preview_size, controller.H * preview_size))
+                        draw = ImageDraw.Draw(preview)
+
+                        # Draw each LED
+                        for y in range(controller.H):
+                            for x in range(controller.W):
+                                color = tuple(controller.matrix_data[y, x])
+                                draw.rectangle(
+                                    [x * preview_size, y * preview_size,
+                                     (x + 1) * preview_size - 1, (y + 1) * preview_size - 1],
+                                    fill=color
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)